### PR TITLE
fix: Use provider HC when Eureka enabled

### PR DESCRIPTION
**Pull request by e4r7hbug**
Resolves a FIXME that has needed to be done for now. Now that
scaffolding is in place, use that to set the option.

See also: PSOBAT-2465

### CODE REVIEW

**Comment by ndcampbell** [comment]
Just for clarification, what is the intended behavior from this change?

**Comment by e4r7hbug** [comment]
This fix will check the `Consider only Amazon health` box in the Deployment Stage by default. This way, Spinnaker will look at the ASG Health Check set to EC2. Currently, we need to specify `{"provider_healthcheck": {"Amazon": true}}` separately.

**Comment by e4r7hbug** [comment]
Wait, Sijis pointed out another issue. Will need to create a provider in a dictionary.

**Comment by ndcampbell** [comment]
Ah, what will the provider in a dictionary do?

File: src/foremast/pipeline/construct_pipeline_block.py [hunk 1]
```diff
@@ -25,6 +25,32 @@
...
+
+    for provider, active in providers.items():
+        if active:
+            provider_healthcheck.append(provider.capitalize())
+    LOG.info('Provider healthchecks: {0}'.format(provider_healthcheck))
+
+    if len(provider_healthcheck) > 0:
+        has_provider_healthcheck = True
+
+    return provider_healthcheck, has_provider_healthcheck
```
Comment (sijis) [inline src/foremast/pipeline/construct_pipeline_block.py]: maybe we should use `namedtuples`?
``` python
ProviderHealthcheck = namedtuple('ProviderHealthcheck', ['provider', 'healthcheck'])
return ProviderHealthcheck(provider_healthcheck, has_providerhealthcheck)
```

File: src/foremast/pipeline/construct_pipeline_block.py [hunk 2]
```diff
@@ -83,22 +109,14 @@ def construct_pipeline_block(env='',
...
-    provider_healthcheck = []
-    for provider, active in settings['asg']['provider_healthcheck'].items():
-        if active:
-            provider_healthcheck.append(provider.capitalize())
-    LOG.info('Provider healthchecks: {0}'.format(provider_healthcheck))
-
-    has_provider_healthcheck = False
-    if len(provider_healthcheck) > 0:
-        has_provider_healthcheck = True
+    provider_healthcheck, has_provider_healthcheck = check_provider_healthcheck(settings)
```
Comment (sijis) [inline src/foremast/pipeline/construct_pipeline_block.py]: Assuming we go with `namedtuple`, we have to adjust this to something like:

``` python
provider_healthcheck = check_provider_healthcheck(settings)
```

**Comment by sijis** [comment]
:+1:

**Comment by ndcampbell** [comment]
LGTM 👍

**Comment by e4r7hbug** [comment]
@ndcampbell: The Provider dictionary was stubbing for future. We only have Amazon right now, but we don't know what the checkbox name will be for Azure or Google.

**Pull request merged**